### PR TITLE
[NOMERGE] rmlint: backport bugfix #438.

### PR DESCRIPTION
**Pull request by Sarkasper**
https://github.com/sahib/rmlint/issues/438
https://github.com/sahib/rmlint/pull/444

### CODE REVIEW

File: srcpkgs/rmlint/patches/444.patch [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+diff --git a/lib/cfg.c b/lib/cfg.c
+index 92b02136..e46756e2 100644
+--- a/lib/cfg.c
++++ b/lib/cfg.c
+@@ -109,7 +109,7 @@ guint rm_cfg_add_path(RmCfg *cfg, bool is_prefd, const char *path) {
+     int rc = 0;
+ 
+ #if HAVE_FACCESSAT
+-    rc = faccessat(AT_FDCWD, path, R_OK, AT_EACCESS|AT_SYMLINK_NOFOLLOW);
++    rc = faccessat(AT_FDCWD, path, R_OK, AT_EACCESS);
```
Comment (ericonr) [inline srcpkgs/rmlint/patches/444.patch:20]: I'd wait for upstream to bless the patch, `AT_SYMLINK_NOFOLLOW` is very specific behavior.

File: srcpkgs/rmlint/template [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Template file for 'rmlint'
 pkgname=rmlint
 version=2.10.1
-revision=2
+revision=4
```
Comment (ericonr) [inline srcpkgs/rmlint/template:4]: 3

File: srcpkgs/rmlint/template [hunk 3]
```diff
@@ -12,6 +12,7 @@ homepage="http://rmlint.rtfd.org/"
 changelog="https://raw.githubusercontent.com/sahib/rmlint/master/CHANGELOG.md"
 distfiles="https://github.com/sahib/${pkgname}/archive/v${version}.tar.gz"
 checksum=10e72ba4dd9672d1b6519c0c94eae647c5069c7d11f1409a46e7011dd0c6b883
+patch_args="-Np1"
```
Comment (ericonr) [inline srcpkgs/rmlint/template:15]: Just remove the prefix from the patch: `a/lib/cfg.c -> lib/cfg.c`, then this isn't necessary.

**Comment by ericonr** [comment]
Also, please follow commit message style: `rmlint: backport bug fix` or something like it

**Comment by Sarkasper** [comment]
> Also, please follow commit message style: `rmlint: backport bug fix` or something like it

Should I add something like [NOMERGE] to the commit message?

**Comment by ericonr** [comment]
No, that type of warning only goes in the PR title itself

File: srcpkgs/rmlint/patches/444.patch (hunk 1 above)
Comment (Sarkasper) [reply srcpkgs/rmlint/patches/444.patch:20]: Ok, thanks for giving input on the issue.

**Comment by Sarkasper** [comment]
Also had some trouble finding an example for a correct commit message in the manual, is there somewhere else I could have found it?

What is the difference between the commit message and PR title?

**Comment by ericonr** [comment]
Oh wow this is completely broken on musl right now... I'm strongly considering simply reverting it to the previous version.

Re. commit messages, see https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#committing-your-changes

**Comment by Sarkasper** [comment]
> Re. commit messages, see https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#committing-your-changes

Ah I should have checked all the examples better, read "without a version change" as in excluding revision change.

**Comment by Sarkasper** [comment]
> Oh wow this is completely broken on musl right now... I'm strongly considering simply reverting it to the previous version.

Yes I has been for quite a while, if I remember correctly downgrading didn't solve it, maybe should have tried couple versions back.

**Comment by ericonr** [comment]
Downgrading with a local package wouldn't be possible, only this version was built with python3.9. Building the previous one works well enough tho.

**Comment by Sarkasper** [comment]
Cool thanks for checking!

**Comment by ericonr** [comment]
If you want to go forward with this PR (instead of a simple revert to the previous version), I think it would be best to revert the entire https://github.com/sahib/rmlint/commit/5ca6d25905cd372e8b07ecd9696a097acb0d5742 commit

**Comment by ChrisBaker97** [comment]
> > Oh wow this is completely broken on musl right now... I'm strongly considering simply reverting it to the previous version.
> 
> Yes I has been for quite a while, if I remember correctly downgrading didn't solve it, maybe should have tried couple versions back.

2.10.0 was where the breaking change occurred, so 2.9.0 was the last working version.

**Comment by ericonr** [comment]
@Sarkasper https://github.com/sahib/rmlint/commit/a726d0cbf839d7a0f35962d67e0f60054d0364f6 is apparently the proper fix, so I'm going to make a PR for that instead. Is that ok for you?

**Comment by Sarkasper** [comment]
> @Sarkasper [sahib/rmlint@a726d0c](https://github.com/sahib/rmlint/commit/a726d0cbf839d7a0f35962d67e0f60054d0364f6) is apparently the proper fix, so I'm going to make a PR for that instead. Is that ok for you?

That's ok for me, thanks!

**Comment by Sarkasper** [comment]
@ericonr I'm unable to install the update, get the same error when installing from repo or local built package:
`[*] Verifying package integrity
rmlint-2.10.1_3: verifying SHA256 hash...
ERROR: rmlint-2.10.1_3: SHA256 hash is not valid: Result not representable`

**Comment by ericonr** [comment]
@Sarkasper can you open a separate issue for this, please? I just tried it and couldn't reproduce (on x86_64, musl or glibc).

**Pull request closed without merging**